### PR TITLE
[NETBEANS-2981] Add script paths to code analysis customizer panels

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-2981

- Add `AnalysisDefaultDocumentListener` class to avoid duplicating code
- Add `AnalysisUiUtils` class to avoid duplicating code (Browse and Search files)

I added `AnalysisDefaultDocumentListener` class because there were many similar implementations of `DocumentListener`. I hope there are no problems.

##### Before

netbeans-2981-code-sniffer-before

##### After

netbeans-2981-code-sniffer-after

##### Before

netbeans-2981-coding-standards-fixer-before

##### After

netbeans-2981-coding-standards-fixer-after

##### Before

netbeans-2981-mess-detector-before

##### After

netbeans-2981-mess-detector-after

##### Before

netbeans-2981-phpstan-before

##### After

netbeans-2981-phpstan-after

### CODE REVIEW

**Comment by junichi11** [comment]
@tmysik Could you please review it when you can make time? Need not hurry. Next week is fine too :)
I committed the changes per analyzer.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/ValidatorCodeSnifferParameter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.analysis.options;
+
+import org.netbeans.modules.php.analysis.ui.analyzer.CodeSnifferCustomizerPanel;
+import org.netbeans.modules.php.analysis.ui.options.CodeSnifferOptionsPanel;
+
+public final class ValidatorCodeSnifferParameter {
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/ValidatorCodeSnifferParameter.java:24]: Perhaps missing `@Nullable`? On fields and/or getters...

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/ValidatorCodingStandardsFixerParameter.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.analysis.options;
+
+import org.netbeans.modules.php.analysis.ui.analyzer.CodingStandardsFixerCustomizerPanel;
+import org.netbeans.modules.php.analysis.ui.options.CodingStandardsFixerOptionsPanel;
+import org.netbeans.modules.php.api.util.StringUtils;
+
+public final class ValidatorCodingStandardsFixerParameter {
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/options/ValidatorCodingStandardsFixerParameter.java:25]: The same - missing `@Nullable`?

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/CodingStandardsFixerCustomizerPanel.java [hunk 3]
```diff
@@ -169,10 +194,35 @@ private void setOptions() {
...
+        );
+    }
+
+    private boolean validateData() {
+        ValidatorCodingStandardsFixerParameter param = ValidatorCodingStandardsFixerParameter.create(this);
+        ValidationResult result = new AnalysisOptionsValidator()
+                .validateCodingStandardsFixer(param)
+                .getResult();
+        if (result.hasErrors()) {
+            context.setError(result.getErrors().get(0).getMessage());
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/CodingStandardsFixerCustomizerPanel.java:217]: One day, we could add a method `getFirstError()` (and `getFirstWarning()`).

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/PHPStanCustomizerPanel.java [hunk 4]
```diff
@@ -221,6 +216,10 @@ private void initComponents() {
         phpStanLevelComboBox = new javax.swing.JComboBox<>();
         phpStanMemoryLimitLabel = new javax.swing.JLabel();
         phpStanMemoryLimitTextField = new javax.swing.JTextField();
+        phpStanLabel = new javax.swing.JLabel();
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/analyzer/PHPStanCustomizerPanel.java:219]: Could we avoid using FQN here in this panel, please? It makes the code more readable, I believe. Thank you.

File: php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/options/CodingStandardsFixerOptionsPanel.java [hunk 5]
```diff
@@ -304,20 +301,21 @@ public void mousePressed(java.awt.event.MouseEvent evt) {
...
-                        .addComponent(codingStandardsFixerSearchButton))
                     .addGroup(layout.createSequentialGroup()
                         .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                             .addComponent(codingStandardsFixerVersionComboBox, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                             .addComponent(codingStandardsFixerConfigComboBox, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                             .addComponent(codingStandardsFixerLevelComboBox, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
                             .addComponent(codingStandardsFixerHintLabel))
-                        .addGap(0, 0, Short.MAX_VALUE))))
+                        .addGap(0, 0, Short.MAX_VALUE))
+                    .addGroup(javax.swing.GroupLayout.Alignment.TRAILING, layout.createSequentialGroup()
```
Comment (tmysik) [inline php/php.code.analysis/src/org/netbeans/modules/php/analysis/ui/options/CodingStandardsFixerOptionsPanel.java:311]: Again, FQN used in this panel...

**Comment by junichi11** [comment]
@tmysik Thanks a lot! Will fix them.

**Comment by junichi11** [comment]
@tmysik Fixed : https://github.com/apache/netbeans/compare/56264dfdfd487babef518c217908eb7f1dd0de3d..84a3d25fbc3919afb0999c041ae54b284ea5fd0a
Thanks!

**Comment by tmysik** [comment]
@junichi11 Great, thanks a lot!

**Comment by tmysik** [comment]
@junichi11 Should I squash and merge?

**Comment by tmysik** [comment]
@junichi11 BTW are the "custom" script compulsory now? :thinking:  Or if the field is empty. the default script will be used?

**Comment by junichi11** [comment]
@tmysik There were unused imports... sorry.
We can merge since this change should not affect CI. : https://github.com/apache/netbeans/compare/84a3d25fbc3919afb0999c041ae54b284ea5fd0a..23d9fa9ef01a13395448761fe74e0f18eb57ff4e

Let's merge as it is. Maybe, it would be easy to read when someone see the logs.

>  if the field is empty. the default script will be used?

Yes :)

**Pull request merged**